### ISSUE TITLE
DoubleCheck locking in LandblockManager::GetLandblock()

### ISSUE THREAD

**Issue by ddevec** [ACEmulator/ACE#350]
Putting this here so I don't forget about it -- its on a TODO list after solving other concurrency errors.  If someone else wants to take the lead on it though, feel free.

### Bug Report or Feature Request (mark with an `x`)
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
```

Currently LandblockManager makes use of double check locking to populate the landblocks array.  Naive double check locking is a known unsafe paradigm in the Java and C# memory models and should be avoided:

https://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html
http://csharpindepth.com/Articles/General/Singleton.aspx

**Comment by Mogwai-TheFurry** [comment]
So, I can't speak to Java, but we're obviously not using it so irrelevant anecdotal comment is irrelevant.  Additional research concludes this problem has been fixed in Java since 5.0.

I just did my due diligence of research on the double check paradigm and I think the only answer is that it's debated.  [This article](https://www.codeproject.com/Tips/887389/Why-You-Should-Use-the-volatile-Keyword-with-the-D) cited a bunch of dead links, but it at least cited them.

I think the short version of this is that we keep an eye on it, and if it breaks we quickly point our finger at this issue.  I'm hesitant to rely on an article from 2006 to call out deficiencies in the .Net framework that other .Net guys claim is acceptable.  Most agree that just using the volatile keyword would solve the problem for us.

TL;DR - add "volatile", keep an eye on it.

**Comment by Mag-nus** [comment]
We should be good here now after the recent landblock refactoring.
